File: 11기 코딩테스트_230308/연습3/BitXorCondition/BitXorCondition_test.py
import unittest

from src.com.luckydave.algorithm.fastcampus.supereasy.BitXorCondition.BitXorCondition import solution


class CustomTestCase(unittest.TestCase):

    def test_1(self):
        # given
        arr = ['10110', '1010', '11110']

        # when
        result = solution(arr)

        # then
        self.assertEqual(2, result)

    def test_2(self):
        # given
        arr = ['1101']

        # when
        result = solution(arr)

        # then
        self.assertEqual(13, result)

    def test_3(self):
        # given
        arr = []

        # when
        result = solution(arr)

        # then
        self.assertEqual(0, result)

    def test_4(self):
        # given
        arr = ['101010', '10101']

        # when
        result = solution(arr)

        # then
        self.assertEqual(63, result)

    def test_5(self):
        # given
        arr = ['110011', '101010', '111110']

        # when
        result = solution(arr)

        # then
        self.assertEqual(39, result)

    def test_6(self):
        # given
        arr = ['11111', '1010101', '1110111']

        # when
        result = solution(arr)

        # then
        self.assertEqual(61, result)


if __name__ == '__main__':
    unittest.main()

File: 11기 코딩테스트_230308/연습3/BitXorCondition/BitXorCondition_test.js
import assert from 'assert'
import solution from './BitXorCondition.js'

describe('Custom Test', function () {
  it('test 1', function (done) {
    // given
    const arr = ['10110', '1010', '11110']

    // when
    const result = solution(arr)

    // then
    assert.deepStrictEqual(2, result)
    done();
  });

  it('test 2', function (done) {
    // given
    const arr = ['1101']

    // when
    const result = solution(arr)

    // then
    assert.deepStrictEqual(13, result)
    done();
  });

  it('test 3', function (done) {
    // given
    const arr = []

    // when
    const result = solution(arr)

    // then
    assert.deepStrictEqual(0, result)
    done();
  });

  it('test 4', function (done) {
    // given
    const arr = ['101010', '10101']

    // when
    const result = solution(arr)

    // then
    assert.deepStrictEqual(63, result)
    done();
  });

  it('test 5', function (done) {
    // given
    const arr = ['110011', '101010', '111110']

    // when
    const result = solution(arr)

    // then
    assert.deepStrictEqual(39, result)
    done();
  });

  it('test 6', function (done) {
    // given
    const arr = ['11111', '1010101', '1110111']

    // when
    const result = solution(arr)

    // then
    assert.deepStrictEqual(61, result)
    done();
  });
})
